nrf_rpc/doc/usage.rst

```diff
@@ -0,0 +1,147 @@
+There are two ways to prase the response.
+
+First is to provide response handler in parameters of :cpp:func:`nrf_rpc_cmd_send`.
+It will be called before :cpp:func:`nrf_rpc_cmd_send` returns.
+It may be called from a different thread.
+
+Second is to call :cpp:func:`nrf_rpc_cmd_rsp_send` or :cpp:func:`nrf_rpc_cbor_cmd_rsp_send` which have output parameters that will contain the response.
+After parsing it :cpp:func:`nrf_rpc_decoding_done` function must be called to indicate that parsing is done and the buffers holding the response can be released.
+
+Events have no response, so nothing more have to be done after sending it.
+
+Sample command encoder using TinyCBOR API:
+
+.. code-block:: c
+
+	/* Function will remotely increment `input` by one and put the
+	 * result into `output`. Function returns 0 on success or
+	 * non-zero error code.
+	 */
+	int remote_inc(int input, int *output)
+	{
+		int err;
+		CborEncoder *encoder;
+		int result[2];
+		struct nrf_rpc_cbor_alloc_ctx ctx;
+
+		NRF_RPC_CBOR_CMD_ALLOC(ctx, &math_group, encoder, 16,
+				       return NRF_RPC_NO_MEM);
+
+		cbor_encode_int(encoder, input);
+
+		err = nrf_rpc_cbor_cmd_send(&ctx, MATH_COMMAND_INC,
+					    remote_inc_rsp, result);
+
+		if (err == NRF_RPC_SUCCESS) {
+			*output = result[0];
+			err = result[1];
+		}
+
+		return err;
+	}
+
+Above code uses `remote_inc_rsp` function to parse the response.
+Following code shows how this function may look like.
+
+.. code-block:: c
+
+	static int remote_inc_rsp(CborValue *parser, void *handler_data)
+	{
+		CborError cbor_err;
+		int *result = (int *)handler_data;
+
+	 	if (!cbor_value_is_integer(parser)) {
+			goto cbor_error_exit;
+		}
+
+		cbor_err = cbor_value_get_int(parser, &result[0]);
+		if (cbor_err != CborNoError) {
+			goto cbor_error_exit;
+		}
+
+		result[1] = NRF_RPC_SUCCESS;
+		return NRF_RPC_SUCCESS;
+
+	cbor_error_exit:
+		result[1] = NRF_RPC_ERR_INVALID_PARAM;
+		return NRF_RPC_SUCCESS;
+	}
+
+
+Decoders
+========
+
+Decoders are registered with a :c:macro:`NRF_RPC_CMD_DECODER`, :c:macro:`NRF_RPC_CBOR_EVT_DECODER` or similar depending on what kind of decoder it will be.
+Decoders are called automatically when command or event is received with a matching id.
+Commands decoders must also send a response.
+
+Decoder associated with the examples above may be following:
+
+.. code-block:: c
+
+	static int remote_inc_handler(CborValue *packet, void* handler_data)
+	{
+		int input;
+		int output;
+		struct nrf_rpc_cbor_alloc_ctx ctx;
+		CborEncoder *encoder;
+		CborError cbor_err;
+
+		/* Parsing the input */
+
+	 	if (cbor_value_is_integer(parser)) {
+			cbor_err = cbor_value_get_int(packet, &input);
+		} else {
+			cbor_err = CborErrorIO;
+		}
+
+		nrf_rpc_decoding_done();
+
+		if (cbor_err != CborNoError) {
+			return NRF_RPC_ERR_INTERNAL;
+		}
+
+		/* Actual hard work is done in below line */
+
+		output = input + 1;
+
+		/* Encoding and sending the response */
+
+		NRF_RPC_CBOR_RSP_ALLOC(ctx, encoder, 16,
```

Comment:
It's a magic number.

This parameter says how big buffer should be for encoded data. I put there 16, because it is for sure enough to encode `int`. I can do a `#define` before the function, if you think that it will be more clear for someone who reads it.